spec/controllers/facility_controller_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe FacilitiesController, type: :controller do
+
+  context 'when user is not logged in' do
+    setup do 
+      user = User.create()
+      allow(controller).to receive(:authenticate_user!).and_return(false)
```

Comment:
Use the devise test helpers to sign in a user https://github.com/plataformatec/devise#test-helpers